etcdserver/raft.go

```diff
@@ -140,7 +140,7 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 	go func() {
 		defer r.onStop()
 		islead := false
-		isCandidate := false
```

Comment:
probably we can simplify all these, we always wait for the conf change if there is any to be applied no matter what. so we do not care if it is candidate or now.